src/strategies_and_params.jl

```diff
@@ -814,99 +820,3 @@ end
 end
 
 @inline update_shift(::DontUpdate, shift, tnorm, args...) = (shift, false, tnorm)
-
-# """
-#     tnorm = apply_memory_noise!(v, w, s_strat, pnorm, tnorm, shift, dτ)
-# Apply memory noise to `v` according to the shift update strategy `s_strat`
-# returning the updated norm.
-# If the strategy does not require updating the norm, it does nothing and returns
-# `tnorm`.
-# """
-# apply_memory_noise!(v, w, s_strat, pnorm, tnorm, shift, dτ) = tnorm
-#
-# # maybe it would be better to apply this before projecting to threshold
-# function apply_memory_noise!(v, w, s_strat::DoubleLogUpdateMemoryNoise,
-#                              pnorm, tnorm, shift, dτ)
-#     if StochasticStyle(v) ∉ [IsStochasticWithThreshold, IsSemistochastic]
-#         @error "Memory noise not defined for $(StochasticStyle(v)). Use `IsStochasticWithThreshold` or `IsSemistochastic` instead."
-#     end
-#     @unpack noisebuffer = s_strat # extract noisebuffer from shift strategy
-#     r̃ = (pnorm - tnorm)/(dτ*pnorm) + shift # instantaneous noisy correction
-#     push!(noisebuffer, r̃) # add to noisebuffer
-#     r_noise = r̃ - sum(noisebuffer)/length(noisebuffer) # subtract Δ average
-#     lv = localpart(v)
-#     for (add, val) in kvpairs(w)
-#         c̃ = lv[add]
-#         c = c̃ + dτ*r_noise*val
-#         if sign(c̃) == sign(c)
-#             lv[add] = c
-#             # !!! Careful! This could lead to a sign change!!!!
-#         end
-#     end
-#     return norm(v, 1) # MPI sycncronising: total number of psips
-# end
-
-"""
-Abstract type for defining the strategy of projection for fciqmc with
-floating point walker number with [`norm_project`](@ref).
-Implemented strategies:
-
-   * [`NoProjection`](@ref)
-   * [`NoProjectionTwoNorm`](@ref)
-   * [`ThresholdProject`](@ref)
-   * [`ScaledThresholdProject`](@ref)
-"""
-abstract type ProjectStrategy end
-
-"Do not project the walker amplitudes. See [`norm_project`](@ref)."
-struct NoProjection <: ProjectStrategy end
-
-"""
-    NoProjectionAccumulator(accumulator::AbstractDVec)  <: ProjectStrategy
-Do not project the walker amplitudes. Maintain a running sum of the coefficient vector
-at each time step in `accumulator`. Take care as accumulator may overflow!
-See [`norm_project`](@ref).
-"""
-struct NoProjectionAccumulator{DV} <: ProjectStrategy
-    accu::DV
-    NoProjectionAccumulator(dv::DV) where (DV<:AbstractDVec) = new{DV}(dv)
-end
```

Comment:
This reminds me that I used this to insert an accumulator into the code, which would compute a rolling average of the coefficient vector (accumulation code would sit in `norm_project!` and dispatch on this strategy). Is there a way we could do this (or similar) with the new setup?